### PR TITLE
Show tours as folders on TaskLists

### LINKED ISSUE #3591: Route should appear as a "folder" when assigned
See #3552

### PR DESCRIPTION
**Pull request by jimakker**
Fixes #3591

### CODE REVIEW

File: js/shared/src/logistics/redux/selectors.js [hunk 1]
```diff
@@ -72,3 +72,54 @@ export const makeSelectTaskListItemsByUsername = () => {
...
+        return []
+      }
+
+      const items = taskList.itemIds
+        .filter(id => Object.prototype.hasOwnProperty.call(tasks, id)) // a task with this id may be not loaded yet
+        .map(id => tasks[id])
+
+        const standaloneTasks = _.filter(items, task => !belongsToTour(task))
+
+        const toursMap = new Map()
```
Comment (jimakker) [inline js/shared/src/logistics/redux/selectors.js:96]: Can/should be refactored as this code is already used in another place

File: js/shared/src/logistics/redux/selectors.js [hunk 2]
```diff
@@ -72,3 +72,54 @@ export const makeSelectTaskListItemsByUsername = () => {
...
+          tours.push({
+            ...tour,
+            items: tasks,
+          })
+        })
+    
+        return {
+          standaloneTasks,
+          tours,
+          allTasks: items
```
Comment (jimakker) [inline js/shared/src/logistics/redux/selectors.js:121]: We are returning all tasks *again*. Should `standaloneTasks` and `tours` be arrays of ids referencing the tasks in `allTasks`?

`allTasks` is needed to show info about delivered percentage etc

**Review by alexsegura — COMMENTED** [review]
There's a strange behavior when reordering tasks in the assigned section. It's ok to have tours appearing first as we can't do much better, but it seems impossible to reorder standalone tasks properly. 

tour_reorder

File: js/app/dashboard/components/TaskList.js [hunk 3]
```diff
@@ -177,8 +184,30 @@ class TaskList extends React.Component {
                 }) }
                 { ...provided.droppableProps }
               >
+                { _.map(tours, (tour, index) => {
+                  return (
+                    <Draggable key={ `tour-${tour['@id']}` } draggableId={ `tour:${tour['@id']}` } index={ this.props.tours.length + index }>
```
[the remark this answers is not in the document]
Comment (alexsegura) [reply js/app/dashboard/components/TaskList.js]: The indexes should be unique for each draggable, this is what explains the strange dragging behavior. 
It's fixed in 29bac4b51d9773212824c2d5aa87f38f79335a4b

<table>
<tr><td>Without unique indexes</td><td>With unique indexes</td></tr>
<tr><td>
</td>
<td>
</td></tr>
</table>

**Comment by alexsegura** [comment]
We have a problem: we are always rendering tours **above** other tasks, while it's still possible to drag'n'drop them. It generates an odd behavior, where you drag'n'drop a tour, and it would always go back to the top (I didn't even check why it shows (7) during a few seconds). 

tour_odd_behavior

We are not going to change the database design right now, however it should be possible to order a tour after/before standalone tasks. 

<table width="100%">
<tr><td>How it looks like in the UI</td><td>How it looks like in Redux</td></tr>
<tr>
<td><pre>
Task 1
Task 2
Tour 1
  Task 3
  Task 4
  Task 5
Task 6
Task 7
Tour 2
  Task 8
  Task 9
  Task 10
</pre></td>
<td>
<pre>
Task 1
Task 2
Task 3
Task 4
Task 5
Task 6
Task 7
Task 8
Task 9
Task 10
</pre></td>
</tr>
</table>

So basically, the selector should return a "flat" list of items, which tasks and tours "mixed" altogether. 
It's not as simple as creating an array of tasks belonging to the same tour, but it should be doable.
